zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,1705 @@
+                );
+            });
+
+            cb.gate(and::expr(vec![
+                meta.query_fixed(q_usable, Rotation::cur()),
+                not::expr(meta.query_advice(rlp_table.data_type, Rotation::cur())),
+            ]))
+        });
+
+        meta.create_gate("DataType::TxHash (signed transaction)", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            let (_, tindex_eq) = tag_index_cmp_1.expr(meta, None);
+
+            // if tag_index == 1
+            cb.condition(is_sig_s(meta) * tindex_eq, |cb| {
+                // RlpTxTag::RlpLength checks.
+                cb.require_equal(
+                    "tag_index::next == 1",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation::next()),
+                    1.expr(),
+                );
+                cb.require_equal(
+                    "tag::next == RlpLength => value_acc::next == index::cur",
+                    meta.query_advice(rlp_table.value_acc, Rotation::next()),
+                    meta.query_advice(index, Rotation::cur()),
+                );
+
+                // RlpTxTag::Rlp checks.
+                cb.require_equal(
+                    "tag::Rotation(2) == RlpTxTag::Rlp",
+                    meta.query_advice(rlp_table.tag, Rotation(2)),
+                    RlpTxTag::Rlp.expr(),
+                );
+                cb.require_equal(
+                    "last tag is Rlp => value_rlc::Rotation(2) == value_rlc::cur",
+                    meta.query_advice(all_bytes_rlc_acc, Rotation(2)),
+                    meta.query_advice(all_bytes_rlc_acc, Rotation::cur()),
+                );
+                cb.require_equal(
+                    "last tag is Rlp => value_acc::Rotation(2) == value_rlc::Rotation(2)",
+                    meta.query_advice(rlp_table.value_acc, Rotation(2)),
+                    meta.query_advice(all_bytes_rlc_acc, Rotation(2)),
+                );
+                cb.require_equal(
+                    "is_last::Rotation(2) == 1",
+                    meta.query_advice(is_last, Rotation(2)),
+                    1.expr(),
+                );
+            });
+
+            cb.gate(and::expr(vec![
+                meta.query_fixed(q_usable, Rotation::cur()),
+                meta.query_advice(rlp_table.data_type, Rotation::cur()),
+            ]))
+        });
+
+        // Constraints that always need to be satisfied.
+        meta.create_gate("always", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            cb.require_boolean(
+                "is_first is boolean",
+                meta.query_advice(is_first, Rotation::cur()),
+            );
+            cb.require_boolean(
+                "is_last is boolean",
+                meta.query_advice(is_last, Rotation::cur()),
+            );
+            cb.require_boolean(
+                "data_type is boolean",
+                meta.query_advice(rlp_table.data_type, Rotation::cur()),
+            );
```

Comment:
We should change it to `require_in_set()` as this column's type  is not a boolean. Its value is in the set `{TxSign, TxHash}`